--- envsetup.mk	2022-05-03 15:10:57.297233893 +0100
+++ android_build/build/core/envsetup.mk	2022-05-03 14:59:16.230683597 +0100
@@ -76,6 +76,11 @@
   HOST_2ND_ARCH := x86
   HOST_IS_64_BIT := true
 else
+ifneq (,$(findstring aarch64,$(UNAME)))
+  HOST_ARCH := aarch64
+  HOST_2ND_ARCH := arm
+  HOST_IS_64_BIT := true
+else
 ifneq (,$(findstring x86,$(UNAME)))
 $(error Building on a 32-bit x86 host is not supported: $(UNAME)!)
 endif
